From 27fca8a3fc8d6d6610dd7473a7dfb376b19d3b04 Mon Sep 17 00:00:00 2001
From: Dave Gerlach <d-gerlach@ti.com>
Date: Wed, 30 Jul 2014 22:41:56 -0500
Subject: [PATCH 0911/1587] remoteproc: wkup_m3: Add support for am4372 to
 wkup_m3_rproc

Add ti,am4372-wkup-m3 compat flags to driver and expand Kconfig for
am43xx support.

Signed-off-by: Dave Gerlach <d-gerlach@ti.com>
[zou: Original patch taken from
ti-sdk-am335x-evm-08.00.00.00-Linux-x86-Install.bin]
Signed-off-by: Cao Zou <cao.zou@windriver.com>
---
 drivers/remoteproc/Kconfig         |   12 +++++++-----
 drivers/remoteproc/wkup_m3_rproc.c |    1 +
 2 files changed, 8 insertions(+), 5 deletions(-)

diff --git a/drivers/remoteproc/Kconfig b/drivers/remoteproc/Kconfig
index 9bedeba..7ed5522 100644
--- a/drivers/remoteproc/Kconfig
+++ b/drivers/remoteproc/Kconfig
@@ -42,16 +42,18 @@ config STE_MODEM_RPROC
 	  If unsure say N.
 
 config WKUP_M3_RPROC
-	bool "AM33xx wkup-m3 remoteproc support"
+	bool "AMx3xx wkup-m3 remoteproc support"
+	depends on SOC_AM33XX || SOC_AM43XX
 	select REMOTEPROC
 	select MAILBOX
 	select OMAP2PLUS_MBOX
 	help
-	  Say y here to support AM33xx wkup-m3.
+	  Say y here to support AMx3xx wkup-m3.
+
+	  Required for Suspend-to-ram and CPUIdle on AM33xx and AM43xx.
+	  Allows for loading of firmware and communication with CM3 PM
+	  coprocessor that is present on AMx3xx family of SoCs
 
-	  Required for Suspend-to-ram and CPUIdle on AM33xx. Allows for
-	  loading of firmware and communication with CM3 PM coprocessor
-	  that is present on AM33xx family of SoCs
 	  If unsure say N.
 
 config DA8XX_REMOTEPROC
diff --git a/drivers/remoteproc/wkup_m3_rproc.c b/drivers/remoteproc/wkup_m3_rproc.c
index 3c135fb..e66b77d 100644
--- a/drivers/remoteproc/wkup_m3_rproc.c
+++ b/drivers/remoteproc/wkup_m3_rproc.c
@@ -449,6 +449,7 @@ static const struct dev_pm_ops wkup_m3_rproc_pm_ops = {
 
 static const struct of_device_id wkup_m3_rproc_of_match[] = {
 	{ .compatible = "ti,am3353-wkup-m3", .data = NULL, },
+	{ .compatible = "ti,am4372-wkup-m3", .data = NULL, },
 	{},
 };
 
-- 
1.7.5.4

